lms/services/exceptions.py

```diff
@@ -27,32 +24,25 @@ class LTIOAuthError(LTILaunchVerificationError):
     """Raised when OAuth signature verification of a launch request fails."""
 
 
-class HAPIError(HTTPInternalServerError):  # pylint: disable=too-many-ancestors
+class HAPIError(ServiceError):
     """
     A problem with an h API request.
 
     This exception class is raised whenever an h API request times out or when
     an unsuccessful, invalid or unexpected response is received from the h API.
 
-    An HAPIError is a 500 Internal Server Error response (HTTPInternalServerError
-    subclass) rather than, say, a 502 Bad Gateway because Cloudflare intercepts
-    gateway error responses and replaces our error page with its own error
-    page, and we don't want that.
-
-    Any arguments or keyword arguments other than ``response`` are passed
-    through to HTTPInternalServerError.
-
     :param response: The response from the HTTP request to the h API
     :type response: requests.Response
     """
 
-    def __init__(self, *args, response=None, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, explanation=None, response=None):
+        super().__init__()
+        self.explanation = explanation
         self.response = response
 
     def __str__(self):
         if self.response is None:
-            return super().__str__()
+            return self.explanation
 
         # Log the details of the h API response. This goes to both Sentry and
```

Comment:
The exceptions get stringified and sent to both Sentry and Papertrail, so adding info into the string representation of an exception is what adds that info to the logs. The actual sending to Sentry and Papertrail is done by those services integrations, not by our own code (for example Sentry automatically reports any non-HTTP exceptions that get raised).

If the exception were caught by the view and either swallowed completely, or wrapped in something other than an `HTTPInternalServerError` (for example wrapped in an `HTTPBadRequest`) that would prevent it from being sent to Sentry, but it would still get logged to Papertrail.